### PR TITLE
Add ping-message-all header with global enums

### PR DESCRIPTION
**Pull request by patrickelectric**
Needs: https://github.com/bluerobotics/ping-protocol/pull/159

Signed-off-by: Patrick José Pereira <patrickelectric@gmail.com>

### CODE REVIEW

File: src/generate/templates/ping-message-all.h.in [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        UNKNOWN = 0,
+        PING1D = 1,
+        PING360 = 2,
+    }
+#ifdef QT_CORE_LIB
+    Q_ENUM_NS(PingMessageId)
+#endif
+}
+
+namespace PingMessageIdNamespace {
```
Comment (jaxxzer) [inline src/generate/templates/ping-message-all.h.in]: Do you really prefer using

`PingMessageIdNamespace::PingMessageId::PING1D_PROFILE`
over
`Ping1dId::PROFILE`?

File: src/generate/templates/ping-message-all.h.in [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+{% for definition in global %}
+#include "ping-message-{{definition}}.h"
+{% endfor %}
+
+namespace PingDeviceTypeNamespace {
+#ifdef QT_CORE_LIB
+    Q_NAMESPACE
+#endif
+    enum PingMessageId {
```
Comment (jaxxzer) [inline src/generate/templates/ping-message-all.h.in]: PingDeviceType

File: src/generate/templates/ping-message-all.h.in [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+// DO NOT EDIT
+//~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!~!
+
+#pragma once
+
+{% for definition in global %}
+#include "ping-message-{{definition}}.h"
+{% endfor %}
+
+namespace PingDeviceTypeNamespace {
```
Comment (jaxxzer) [inline src/generate/templates/ping-message-all.h.in]: If you need the namespace and QNamespace (I don't think you do), I think you should consider committing this declaration to ping-viewer.

The enum nested inside the namespace seems unnecessarily excessive to the point it complicates usage.

I think it's good to have this enum, but I prefer just the enum or namespace. The same request applies to the PingMessageId enum.

**Comment by patrickelectric** [comment]
Fixed with enum class.

**Pull request merged**